pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -937,6 +944,45 @@ func (r ReconcileHyperConverged) migrateCdiConfigurations(req *common.HcoRequest
 	return adoptCdiConfigs(req, cdi.Spec.Config), nil
 }
 
+func (r ReconcileHyperConverged) migrateImsConfigurations(req *common.HcoRequest) (bool, error) {
+	req.Logger.Info("read IMS configmap")
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imsCmName,
+			Namespace: req.Namespace,
+		},
+	}
+
+	if err := hcoutil.GetRuntimeObject(req.Ctx, r.client, cm, req.Logger); err != nil {
+		if apierrors.IsNotFound(err) {
+			req.Logger.Info("IMS configmap already removed")
+			return false, nil
+		}
+		req.Logger.Info("failed to get IMS configmap", "error", err.Error())
+		return false, err
+	}
+
+	modified := false
+	vddkInitImage, ok := cm.Data[vddkInitImakeKey]
+	if !ok {
+		// if the user never set the value we should simply continue
+		req.Dirty = false || req.Dirty
+		modified = false
+	} else {
+		if *req.Instance.Spec.VddkInitImage != vddkInitImage {
+			req.Logger.Info("updating the HyperConverged CR from the IMS configMap")
+			req.Instance.Spec.VddkInitImage = &vddkInitImage
+			req.Dirty = true
+			modified = true
+		} else {
```

Comment:
There is no effet for this else - please drop it

Reply:
done